web/client/utils/MapUtils.js

```diff
@@ -411,6 +412,38 @@ const parseLayoutValue = (value, size = 0) => {
     return isNumber(value) ? value : 0;
 };
 
+const prepareMapObjectToCompare = obj => {
+    const skippedKeys = ['apiKey', 'time', 'args', 'fixed'];
+    const shouldBeSkipped = (key) => skippedKeys.reduce((p, n) => p || key === n, false);
+    Object.keys(obj).forEach(key => {
+        const value = obj[key];
+        const type = typeof value;
+        if (type === "object" && value !== null && !shouldBeSkipped(key)) {
+            prepareMapObjectToCompare(value);
+            if (!Object.keys(value).length) {
+                delete obj[key];
+            }
+        } else if (type === "undefined" || !value || shouldBeSkipped(key)) {
+            delete obj[key];
+        }
+    });
+};
+
+const compareMapChanges = (map1 = {}, map2 = {}) => {
+    const pickedFields = [
+        'map.layers',
+        'map.backgrounds',
+        'map.text_search_config',
```

Comment:
I noticed that I can still not leave this map /#/viewer/openlayers/2464 trying to open and exiting, the map asks me to confirm This is because of the name mismatch between old search configuration name and the new...
Can we do something to pick also old configs for compare?

Reply:
done